api/rooms/views/main.py

```diff
@@ -0,0 +1,22 @@
+import requests, json
+
+from django.views     import View
+from django.http      import JsonResponse
+
+from my_settings import API_KEY
+
+class SearchRoomView(View):
+    def get(self, request):
+        try:
+            search_address = request.GET.get('search_address', None)
+            search_address = search_address.strip()
+            type           = request.GET.get('type')
+            category       = request.GET.get('category')
+           
+            if len(search_address) < 2 : 
+                    return JsonResponse({'message':'ADDRESS_IS_TOO_SHORT'}, status=400)
+            response= requests.get(f'http://api.vworld.kr/req/search?service=search&request=search&version=2.0&crs=EPSG:900913&size=100&page=1&query={search_address}&type={type}&category={category}&format=json&errorformat=json&key={API_KEY}')
+            search_data = response.json()
+            return JsonResponse({'message': 'SUCCESS', 'search_data': search_data}, status=200)
+        except KeyError:
+            return JsonResponse({'message' : 'KEY_ERROR'}, status=400)
```

Comment:
아래와 같이 가독성을 위해서 아래와 같이 수정해보면 어떨까요?

```suggestion
            if len(search_address) < 2 : 
                    return JsonResponse({'message':'ADDRESS_IS_TOO_SHORT'}, status=400)
                    
            response = requests.get(f'http://api.vworld.kr/req/search? 
                service = search & 
                request = search & 
                version = 2.0 &
                crs     = EPSG:900913 &
                size    = 100 &
                page    = 1 &
                query   = {address} & 
                type    = {type} &
                category = {category} &
                format   = json &
                errorformat = json & 
                key = {API_KEY}'
            )
                
            search_data = response.json()
            
            return JsonResponse({'message': 'SUCCESS', 'search_data': search_data}, status=200)
            
        except KeyError:
            return JsonResponse({'message' : 'KEY_ERROR'}, status=400)
```

Reply:
이렇게 수정하고 싶어서 검색했었는데 안나와서... 안되나보다 했는데! 역시 멘토님....👍👍🏻👍🏼👍🏽👍🏾👍🏿
수정하겠습니다